### PR TITLE
Refactor span attribute serialization to use JSON strings

### PR DESCRIPTION
**Pull request by dnandakumar-nv**
## Description
Replaced _to_dict with _to_json_string for safer, OTLP-compatible span attribute serialization. Added robust handling for nested structures, Pydantic models, and edge cases. Introduced comprehensive tests to validate serialization behavior and ensure stability.

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/NeMo-Agent-Toolkit/blob/develop/docs/source/resources/contributing/index.md).
- We require that all contributors "sign-off" on their commits. This certifies that the contribution is your original work, or you have rights to submit it under the same license, or a compatible license.
  - Any contribution which contains commits that are not Signed-Off will not be accepted.
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

## Summary by CodeRabbit

* **Bug Fixes**
  * Span input/output payloads are now serialized and stored as JSON strings for OTLP compatibility, including recursive normalization, extraction of top-level value fields, filtering of None entries, and safe fallbacks for non-serializable data; internal documentation for the transformation was added.

* **Tests**
  * Expanded test coverage for JSON serialization edge cases: strings, dicts/lists, model types, value-field extraction, None filtering, mixed types, and fallback behavior.

### CODE REVIEW

File: packages/nvidia_nat_core/src/nat/observability/exporter/span_exporter.py [hunk 1]
```diff
@@ -323,20 +324,53 @@ def _process_end_event(self, event: IntermediateStep):
...
+            if isinstance(obj, (list, tuple)):
+                return [_normalize(item) for item in obj]
+
+            # Primitives and other objects: return as-is (json.dumps will handle via default=str)
+            return obj
+
+        try:
+            normalized = _normalize(data)
+            return json.dumps(normalized, default=str)
+        except Exception:
```
Comment (mpenn) [inline packages/nvidia_nat_core/src/nat/observability/exporter/span_exporter.py]: Consider a debug or warning log here to help with debugging. For example, something like this:

`logger.debug("Span attribute serialization failed, using str fallback: %s", e)`

**Review by mpenn — APPROVED** [review]
Overall, this looks good. Just one minor nit.

File: packages/nvidia_nat_core/src/nat/observability/exporter/span_exporter.py [hunk 2]
```diff
@@ -323,20 +324,53 @@ def _process_end_event(self, event: IntermediateStep):
...
+            A valid JSON string representation of the data.
+        """
 
-        if 'value' in result and result['value'] is not None:
-            return result['value']
+        def _normalize(obj: typing.Any) -> typing.Any:
+            """Recursively normalize objects for JSON serialization."""
+            # Pydantic models: dump and recursively normalize
+            if hasattr(obj, 'model_dump'):
+                dumped = obj.model_dump(exclude_none=True)
```
Comment (willkill07) [inline packages/nvidia_nat_core/src/nat/observability/exporter/span_exporter.py]: Why not model_dump with mode=json to avoid the recursion?

**Review by willkill07 — APPROVED** [review]
One additional nit in addition to Matt's review
